internal/driver/driver.go

```diff
@@ -652,6 +616,23 @@ func addressAndPort(xaddr string) (string, string) {
 	}
 }
 
+// todo: this should be integrated better with getDeviceInformation to avoid creating another temporary client
```

Comment:
Yes! Why was this not done now?

Reply:
Mostly because it was quick and dirty (which isn't a good excuse :wink: ), but also because that whole client and temporaryOnvifClient concept needs a good refactoring.

Reply:
Will address this todo as part of a future story.